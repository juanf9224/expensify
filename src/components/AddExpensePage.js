import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export const AddExpensePage = (props) => (
  <div>
    <h1>Add Expense</h1>
    <ExpenseForm
      onSubmit={(expense) => {
        console.log(expense);
        props.startAddExpense(expense);
        props.history.push('/')
      }}
    />
  </div>
);

const mapDispatchToProps = (dispatch) => {
  return {
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
  }
}

export default connect(undefined, mapDispatchToProps)(AddExpensePage);