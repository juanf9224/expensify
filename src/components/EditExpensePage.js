import React from 'react';
import { connect } from 'react-redux';
import { editExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export const EditExpensePage = (props) => (
  <div>
    <ExpenseForm
      expense={props.expense}
      onSubmit={(expense) => {
        console.log(expense);
        props.dispatch(editExpense(props.expense.id, expense));
        props.history.push('/');
      }}
    />
  </div>
);

const mapToStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(e => e.id === props.match.params.id)
  }
};

export default connect(mapToStateToProps)(EditExpensePage);