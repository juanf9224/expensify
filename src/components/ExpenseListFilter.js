import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilter extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      calendarFocus: null
    };
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  }

  onFocusChange = (calendarFocus) => {
    this.setState(() => ({ calendarFocus }))
  }

  isOutsideRange = () => {
    return false;
  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={(e) => this.props.dispatch(setTextFilter(e.target.value))}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => this.props.dispatch(e.target.value === 'date' ? sortByDate() : sortByAmount())}
        >
          <option value="amount">
            Amount
      </option>
          <option value="date">
            Date
      </option>
        </select>

        <DateRangePicker     
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocus}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={this.isOutsideRange}          
        />
      </div>
    );

  }
}

const mapToState = (state) => {
  return {
    filters: state.filters
  }
}

export default connect(mapToState)(ExpenseListFilter);