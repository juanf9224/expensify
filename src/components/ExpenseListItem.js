import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ id, description, amount, createdAt, dispatch }) => (
  
  <tr>
    <td>{description}</td>
    <td>{amount}</td>
    <td>{createdAt}</td>
    <td><Link to={`/edit/${id}`}>Edit</Link></td>
    <td><button onClick={() => dispatch(removeExpense({ id }))}>Remove</button></td>
  </tr>
);

export default connect()(ExpenseListItem);
