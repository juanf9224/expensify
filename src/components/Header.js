import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Go To Dashboard Page</NavLink><br />
    <NavLink to="/create" activeClassName="is-active">Go To Add Page</NavLink><br />
    <NavLink to="/edit" activeClassName="is-active">Go Edit Page</NavLink> <br />
    <button onClick={startLogout}>Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch) => {
  return {
    startLogout: () => dispatch(startLogout())
  }
}

export default connect(undefined, mapDispatchToProps)(Header);