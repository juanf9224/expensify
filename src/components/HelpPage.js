import React from 'react';

const HelpPage = () => (
  <div>
    Help Expense component
  </div>
);

export default HelpPage;