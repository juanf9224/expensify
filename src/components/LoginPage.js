import React, { useState } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const onSubmit = (e) => {
    e.preventDefault();
    startLogin();
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="username"
        value={username}
        onChange={e => setUsername(e.target.value)}
      />
      <input
        type="password"
        value={password}
        onChange={e => setPassword(e.target.value)}
      />
      <button type="submit">Login</button>
    </form>
  );
}

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);