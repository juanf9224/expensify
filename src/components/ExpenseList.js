import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => {
  const expenseListItems = props.expenses.map(e => (
    <ExpenseListItem key={e.id} {...e}/>
    )
  );
  return (
    <div>
      <h1>Expense List</h1>
      <table>
        <thead>
          <tr>
            <th>Description</th>
            <th>Amount</th>
            <th>Created At</th>
          </tr>
        </thead>
        <tbody>
          {expenseListItems}
        </tbody>
      </table>
    </div>
  )
};

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
};

export default connect(mapStateToProps)(ExpenseList);