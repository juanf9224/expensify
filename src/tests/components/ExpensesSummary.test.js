import { shallow } from 'enzyme';
import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={23500}/>);
  expect(wrapper).toMatchSnapshot();
});