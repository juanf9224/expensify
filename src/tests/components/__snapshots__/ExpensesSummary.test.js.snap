// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $235.00
  </h1>
</div>
`;
